### PR TITLE
feat: docsearch command to search documentation

### PR DESCRIPTION
**Pull request by pdellaert**
feat: doc command to search documentation

## Description
This command has the following functionality:
* Listen to `.doc`, `.docsearch` and `.documentation`
* If no argument is given, it provides a link to the documentation website
* If an argument is given (one or more words), it provides a link to the documentation search functionality
* It checks for profanity
* It checks for URLs to minimize the risk of abusing the bot for pushing your own URL into a bot output
* It URI encodes the search to prevent injection of unwanted characters in the URL itself

## Test Results
Positive cases
Screen Shot 2022-08-09 at 6 29 32 PM

Negative cases
Screen Shot 2022-08-09 at 6 29 44 PM

## Discord Username
straks#7240

### CODE REVIEW

**Comment by NUTTTTTTTT** [comment]
Great PR, had a good test and check, all looks good to me but I'll wait for someone with a little more expertise to have a look over it and approve before merging. 🚀 

image
image

File: src/commands/general/docsearch.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (searchWords.length <= 1) {
+            const noQueryEmbed = makeEmbed({
+                title: 'FlyByWire Documentation',
+                description: 'Find the full FlyByWire Documentation [here](https://docs.flybywiresim.com/).',
+            });
+            return msg.channel.send({ embeds: [noQueryEmbed] });
+        }
+
+        console.log(searchWords);
```
Comment (NathanInnes) [inline src/commands/general/docsearch.ts]: No console logs.

**Comment by pdellaert** [comment]
There was a rogue leftover debug output console.log in there... woops. Removed it.

**Review by oim1 — APPROVED** [review]

File: src/commands/general/docsearch.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import Filter from 'bad-words';
+import { Colors } from 'discord.js';
+import { CommandDefinition } from '../../lib/command';
+import { CommandCategory } from '../../constants';
+import { makeEmbed } from '../../lib/embed';
+
```
Comment (NathanInnes) [inline src/commands/general/docsearch.ts:6]:
```suggestion

const DOCS_BASE_URL = 'https://docs.flybywiresim.com';
```

File: src/commands/general/docsearch.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        // Removing first element as it is an empty string
+        searchWords.shift();
+        const searchQuery = searchWords.join(' ');
+
+        // Safety to prevent users from entering unexpected data that might result in strange behavior in a URL.
+        const encodedSearchQuery = encodeURIComponent(searchQuery);
+
+        const queryEmbed = makeEmbed({
+            title: 'FlyByWire Documentation Search',
+            description: `Search the FlyByWire Documentation for "${searchQuery}" [here](https://docs.flybywiresim.com/?q=${encodedSearchQuery}).`,
```
Comment (NathanInnes) [inline src/commands/general/docsearch.ts:49]:
```suggestion
            description: `Search the FlyByWire Documentation for "${searchQuery}" [here](${DOCS_BASE_URL}/?q=${encodedSearchQuery}).`,
```

File: src/commands/general/docsearch.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                return msg.channel.send({ embeds: [URLEmbed] });
+            } catch (_) { /**/ }
+
+            const filter = new Filter();
+            if (filter.isProfane(searchWord)) {
+                return msg.reply('Please do not use profane language with this command.');
+            }
+        }
+
+        // Removing first element as it is an empty string
```
Comment (NathanInnes) [inline src/commands/general/docsearch.ts:40]: Need this comment?

**Review by NathanInnes — CHANGES_REQUESTED** [review]
Good work with this, see review comments

File: src/commands/general/docsearch.ts (hunk 1 above)
Comment (NathanInnes) [reply src/commands/general/docsearch.ts]: Resolved.

File: src/commands/general/docsearch.ts (hunk 3 above)
Comment (NathanInnes) [reply src/commands/general/docsearch.ts:49]: Also add to other uses?

File: src/commands/general/docsearch.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import Filter from 'bad-words';
+import { Colors } from 'discord.js';
+import { CommandDefinition } from '../../lib/command';
+import { CommandCategory } from '../../constants';
+import { makeEmbed } from '../../lib/embed';
+
+export const docsearch: CommandDefinition = {
+    name: ['docsearch', 'documentation', 'doc'],
```
Comment (NathanInnes) [inline src/commands/general/docsearch.ts:8]: Add 'docs' alias also?

File: src/commands/general/docsearch.ts (hunk 4 above)
Comment (pdellaert) [reply src/commands/general/docsearch.ts:40]: No, just wanted to clarify if anyone was wondering. Happy to remove it

File: src/commands/general/docsearch.ts (hunk 2 above)
Comment (pdellaert) [reply src/commands/general/docsearch.ts:6]: Should be resolved.

File: src/commands/general/docsearch.ts (hunk 3 above)
Comment (pdellaert) [reply src/commands/general/docsearch.ts]: Should be resolved.

File: src/commands/general/docsearch.ts (hunk 5 above)
Comment (pdellaert) [reply src/commands/general/docsearch.ts]: Should be resolved

File: src/commands/general/docsearch.ts (hunk 4 above)
Comment (pdellaert) [reply src/commands/general/docsearch.ts]: Removed it, so should be resolved.

**Comment by pdellaert** [comment]
Thanks @oim1, @NathanInnes and @NUTTTTTTTT for the review, comments and suggestions. Everything should be addressed, but if anything else stands out, happy to improve it :)

**Review by NathanInnes — APPROVED** [review]

**Pull request merged**